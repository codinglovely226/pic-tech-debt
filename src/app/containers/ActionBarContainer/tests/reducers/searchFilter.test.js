/* eslint-disable */

import searchFilterReducer from '../../reducers/searchFilter';

import {
  addCallRecord,
  addCase,
} from 'containers/CaseEntry/actions/case';

import {
  toggleNewCallModal,
  updateNewCallModalCallerLocation,
  updateNewCallModalCallerOrganisation,
  updateNewCallModalCentreName,
  updateNewCallModalFilter,
  updateNewCallModalId,
  updateNewCallModalName,
  updateNewCallModalPostcode,
  updateNewCallModalSelected,
  updateNewCallModalSex,
  updateNewCallModalTime,
} from '../../actions';


describe('searchFilterReducer', () => {
  var mockedDate = new Date(2018, 2, 4, 4, 59, 13, 805);
  global.Date = jest.fn(() => mockedDate);
  it('returns the initial state', () => {
    expect(searchFilterReducer(undefined, {})).toMatchSnapshot();
  });
  it('add case', () => {
    const action = addCase({ 'name': 'Test SPI' }, 'WAPIC');
    action.id = 'b72540f9163ef35a05213611e012f2bdf8ccefb3';
    expect(searchFilterReducer({}, action)).toMatchSnapshot();
  });
  it('add call record', () => {
    expect(searchFilterReducer({}, addCallRecord(1, { 'name': 'Test SPI' }))).toMatchSnapshot();
  });
  it('toggle new call modal', () => {
    expect(searchFilterReducer({}, toggleNewCallModal())).toMatchSnapshot();
  });
  it('update new call modal caller location', () => {
    expect(searchFilterReducer({}, updateNewCallModalCallerLocation('New location'))).toMatchSnapshot();
  });
  it('update new call modal caller organisation', () => {
    expect(searchFilterReducer({}, updateNewCallModalCallerOrganisation('New organisation'))).toMatchSnapshot();
  });
  it('update new call modal centre name', () => {
    expect(searchFilterReducer({}, updateNewCallModalCentreName('New centre name'))).toMatchSnapshot();
  });
  it('update new call modal filter', () => {
    expect(searchFilterReducer({}, updateNewCallModalFilter('New filter'))).toMatchSnapshot();
  });
  it('update new call modal id', () => {
    expect(searchFilterReducer({}, updateNewCallModalId('New ID'))).toMatchSnapshot();
  });
  it('update new call modal name', () => {
    expect(searchFilterReducer({}, updateNewCallModalName('New name'))).toMatchSnapshot();
  });
  it('update new call modal postcode', () => {
    expect(searchFilterReducer({}, updateNewCallModalPostcode('New postcode'))).toMatchSnapshot();
  });
  it('update new call modal selected', () => {
    expect(searchFilterReducer({}, updateNewCallModalSelected('New selected'))).toMatchSnapshot();
  });
  it('update new call modal sex', () => {
    expect(searchFilterReducer({}, updateNewCallModalSex('New sex'))).toMatchSnapshot();
  });
  it('update new call modal time', () => {
    expect(searchFilterReducer({}, updateNewCallModalTime('New time'))).toMatchSnapshot();
  });
});
