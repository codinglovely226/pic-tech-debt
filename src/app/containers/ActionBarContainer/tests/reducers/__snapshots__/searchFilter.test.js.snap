exports[`searchFilterReducer add call record 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer add case 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer returns the initial state 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer toggle new call modal 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": true,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal caller location 1`] = `
Object {
  "callerLocation": "New location",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal caller organisation 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "New organisation",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal centre name 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "New centre name",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal filter 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "New filter",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal id 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "New ID",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal name 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "New name",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal postcode 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "New postcode",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal selected 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": "New selected",
  "sex": "",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal sex 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "New sex",
  "show": false,
  "time": "",
}
`;

exports[`searchFilterReducer update new call modal time 1`] = `
Object {
  "callerLocation": "",
  "callerOrganisation": "",
  "centreName": "",
  "filter": "all",
  "name": "",
  "naturalId": "",
  "postcode": "",
  "selected": null,
  "sex": "",
  "show": false,
  "time": "New time",
}
`;
